flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/BuiltInFunctionTestBase.java

```diff
@@ -383,18 +439,69 @@ public String toString() {
         }
     }
 
-    private static class SqlErrorTestItem extends ErrorTestItem {
-        final String expression;
+    private static class SqlErrorTestItem extends ErrorTestItem<String> {
 
         private SqlErrorTestItem(
                 String expression, String errorMessage, boolean expectedDuringValidation) {
-            super(errorMessage, expectedDuringValidation);
-            this.expression = expression;
+            super(expression, errorMessage, expectedDuringValidation);
+        }
+
+        @Override
+        String expression() {
+            return expression;
         }
 
         @Override
         public String toString() {
             return "[SQL] " + expression;
         }
     }
+
+    private static List<DataType> createDataTypes(
+            DataTypeFactory dataTypeFactory, AbstractDataType<?>[] dataTypes) {
+        return Stream.of(dataTypes)
+                .map(dataTypeFactory::createDataType)
+                .collect(Collectors.toList());
+    }
+
+    /** Helper POJO to store test parameters. */
+    public static class TableTestSpecColumn {
```

Comment:
`Column` sounds very internal to this class. Tests don't know columns but only test specs. How about `ResultSpec` to keep it short?

Reply:
Makes sense, thx, couldn't come up with a nice name.